File: lektionen/javascript/Kata/k5_maximum_subarray_sum.js
var maxSequence = function(arr){
  if (arr.length == 0) {
    return 0;
  }
  else if (arr.every(x => x >= 0)) {
    return arr.reduce((a,b) => a + b);
  }
  else if (arr.every(x => x < 0)) {
    return 0;
  }
  else {
    var result = 0;
    for (i = 0; i <= arr.length - 2; i++){
      for(j = arr.length; j >= i + 2 ; j--) {
        if (result == 0 || result < arr.slice(i, j).reduce((a,b) => a + b)) {
          result = arr.slice(i, j).reduce((a, b) => a + b);
        }
      }
    }
    arr.forEach(x => {
      if (x > result) {
        result = x;
      }
    });  
  return result;
  }
}

File: lektionen/python/Kata/k5_maximum_subarray_sum.py
def max_sequence(arr):
  if len(arr) == 0:
    return 0
    
  def all_neg(arr):
    for e in arr:
      if e > 0:
        return False
    return True

  if all_neg(arr):
    return 0

  total = 0
  for i in range(len(arr)):
    for j in range(i+1, len(arr) + 1):
      if sum(arr[i:j]) > total:
        total = sum(arr[i:j])

  return total


print(max_sequence([7, 4, 11, -11, 39, 36, 10, -6, 37, -10, -32, 44, -26, -34, 43, 43]))